File: backend/app/schemas/company.py
from __future__ import annotations

from pydantic import BaseModel


class CompanySummary(BaseModel):
    id: int
    cnpj: str
    razao_social: str
    nome_fantasia: str | None
    apelido_pasta: str | None
    inscricao_estadual: str | None
    municipio: str | None
    uf: str | None
    active: bool
    regime_label: str


class CompanyListResponse(BaseModel):
    items: list[CompanySummary]


class CompanySummaryKpis(BaseModel):
    obligations_total: int
    delivered_total: int
    pending_total: int
    divergences_total: int
    evidences_total: int
    installments_total: int


class CompanyObligationPreview(BaseModel):
    obligation_code: str
    obligation_name: str
    status: str
    department: str
    source: str | None
    due_date: str | None
    delivered_at: str | None


class CompanyDetailResponse(BaseModel):
    company: CompanySummary
    period: str
    cnpj: str
    inscricao_estadual_display: str
    municipio_uf: str
    regime_label: str
    kpis: CompanySummaryKpis
    obligations: list[CompanyObligationPreview]
    evidences_preview: int
    divergences_preview: int

File: frontend/src/types/company.ts
export type CompanySummary = {
  id: number;
  cnpj: string;
  razao_social: string;
  nome_fantasia: string | null;
  apelido_pasta: string | null;
  inscricao_estadual: string | null;
  municipio: string | null;
  uf: string | null;
  active: boolean;
  regime_label: string;
};

export type CompanyListResponse = {
  items: CompanySummary[];
};

export type CompanySummaryKpis = {
  obligations_total: number;
  delivered_total: number;
  pending_total: number;
  divergences_total: number;
  evidences_total: number;
  installments_total: number;
};

export type CompanyObligationPreview = {
  obligation_code: string;
  obligation_name: string;
  status: string;
  department: string;
  source: string | null;
  due_date: string | null;
  delivered_at: string | null;
};

export type CompanyDetailResponse = {
  company: CompanySummary;
  period: string;
  cnpj: string;
  inscricao_estadual_display: string;
  municipio_uf: string;
  regime_label: string;
  kpis: CompanySummaryKpis;
  obligations: CompanyObligationPreview[];
  evidences_preview: number;
  divergences_preview: number;
};
